proto/pulumi/provider.proto

```diff
@@ -78,7 +78,17 @@ service ResourceProvider {
     // thereof can negatively impact the end-user experience, as the provider inputs are using for detecting and
     // rendering diffs.
     rpc CheckConfig(CheckRequest) returns (CheckResponse) {}
-    // DiffConfig checks the impact a hypothetical change to this provider's configuration will have on the provider.
+
+    // `DiffConfig` compares an existing ("old") provider configuration with a new configuration and computes the
+    // difference (if any) between them. `DiffConfig` is to provider resources what [](pulumirpc.ResourceProvider.Diff)
+    // is to individual resources. `DiffConfig` should only be called values that have at some point been validated by a
```

Comment:
"should only be called with values"? grammar.

Reply:
It is worth mentioning here that DiffConfig implementors need to be aware of schema changes. In case where the resource or provider schemas are changing, the `olds` passed to DiffRequest might have been written by the previous version of the provider, and the implementaiton needs to take care of transparently migrating to the new schema.

Reply:
Good catch and call out -- will update!